src/librbd/io/ObjectRequest.cc

```diff
@@ -450,6 +459,27 @@ AbstractObjectWriteRequest<I>::AbstractObjectWriteRequest(
 }
 
 template <typename I>
+void AbstractObjectWriteRequest<I>::add_write_hint(
+    librados::ObjectWriteOperation *wr) {
+  I *image_ctx = this->m_ictx;
+  RWLock::RLocker snap_locker(image_ctx->snap_lock);
+  if (image_ctx->enable_alloc_hint &&
```

Comment:
nit: may be no need to check for `image_ctx->enable_alloc_hint` here as it will be checked by `ObjectRequest<I>::add_write_hint()` anyway?